### PR TITLE
Added frontend and backend functionality to retrieve playable game data

### PR DESCRIPTION
**Pull request by macleanb**
Complete -- added backend view/url for retrieving game data for all games that the logged in user can play. Also added frontend getURL_PlayableGames and api_GetPlayableGames functions on frontend that are now called every time a user logs in and is on the Game menu screen. The response from the backend is saved in state "playableGames" in Game.js.

Passes existing tests. Also verified data is being received using console.log. Will add specific tests once the frontend UI functionality is in place.

### CODE REVIEW

File: frontend/cheapchess/src/pages/Game.js [hunk 1]
```diff
@@ -174,6 +177,24 @@ const Game = () =>
...
+     game menu is requested from FormManager */
+  useEffect(() => {
+    if (
+      appState?.imports &&
+      auth?.status === constants.STATUS_AUTHENTICATED &&
+      formMode === constants.FORM_MODE_GAME_NEW_CONTINUE &&
+      formType === constants.FORM_TYPE_GAME_MENU
+      )
+    {
+      appState.imports.getPlayableGames().then((result) => {
```
[the remark this answers is not in the document]
Comment (chsbca) [reply frontend/cheapchess/src/pages/Game.js:190]: this is more so to catch errors just in case something fails here or from the backend

**Review by chsbca — COMMENTED** [review]
It looks good to me. I just added some suggestions for catching errors and a possibility of a loading state/screen, but we can save that for later.

File: frontend/cheapchess/src/pages/Game.js [hunk 2]
```diff
@@ -174,6 +177,24 @@ const Game = () =>
...
+  /* Get playable games data once user is logged in and 
+     game menu is requested from FormManager */
+  useEffect(() => {
+    if (
+      appState?.imports &&
+      auth?.status === constants.STATUS_AUTHENTICATED &&
+      formMode === constants.FORM_MODE_GAME_NEW_CONTINUE &&
+      formType === constants.FORM_TYPE_GAME_MENU
+      )
+    {
```
[the remark this answers is not in the document]
Comment (macleanb) [reply frontend/cheapchess/src/pages/Game.js:189]: Yes I think that would be a good idea in the UI component for selecting existing games (dropdown, select box, etc.)

**Comment by macleanb** [comment]
Great feedback.  I believe I resolved all the issues and the tests still pass.

**Review by chsbca — APPROVED** [review]

**Review by chsbca — APPROVED** [review]

**Pull request merged**